corehq/apps/userreports/pillow.py

```diff
@@ -49,6 +49,31 @@ def timed(*args, **kw):
     return timed
 
 
+def _filter_by_hash(configs, ucr_division):
+    ucr_start = ucr_division[0]
+    ucr_end = ucr_division[-1]
+    filtered_configs = []
+    for config in configs:
+        table_hash = hashlib.md5(config.table_id).hexdigest()[0]
+        if ucr_start <= table_hash <= ucr_end:
+            filtered_configs.append(config)
+    return filtered_configs
+
+
+def _exclude_missing_domains(configs):
+    from corehq.apps.es import DomainES
+    from corehq.elastic import ESError
+
+    config_domains = {conf.domain for conf in configs}
```

Comment:
I was going to do this by adding a `server_environment` variable to the static config and filtering on that, but this looks good for now. 

Could you limit this to only being filtered for static reports? This filter isn't useful for dynamic reports and I think this might not work well on prod, because I think there are some issues when you feed a large array to ES. @benrudolph you ran into that in reports right?